libs/core/langchain_core/prompts/chat.py

```diff
@@ -66,6 +67,17 @@ def format_messages(self, **kwargs: Any) -> List[BaseMessage]:
             List of BaseMessages.
         """
 
+    async def aformat_messages(self, **kwargs: Any) -> List[BaseMessage]:
+        """Format messages from kwargs. Should return a list of BaseMessages.
+
+        Args:
+            **kwargs: Keyword arguments to use for formatting.
+
+        Returns:
+            List of BaseMessages.
+        """
+        return await run_in_executor(None, self.format_messages, **kwargs)
```

Comment:
This is another place where I feel like we need to delegate to the sync implementation.

I'm a bit worried about having this dichotomy / how to make it obvious that this is is what's going on

Reply:
cc @baskaryan  / @nfcampos 

Do we keep run_in_executor for consistency or use a blocking call to the sync code?

Reply:
Another PR where this question comes up: https://github.com/langchain-ai/langchain/pull/19652/files

Reply:
This is the part that makes it all weird:

https://github.com/langchain-ai/langchain/blob/2e0ddd6fb805af53d9bfb0edc463543a0312530f/libs/core/langchain_core/prompts/chat.py#L934-L934

Chat prompt templates are constructed of multiple messages each of which can be formatted independently -- this 
would mean that interpolating a few variables into essentially a list of dicts we're launching several threads

Reply:
Another possibility if we don't want to choose now is to raise NotImplementedError in the default interface.

Reply:
I intend to implement all subclasses in the end so the default interface won't be used anyway.

Reply:
> a list of dicts we're launching several threads

Actually, it's not launching several threads, it's submitting tasks to the default thread-pool (which has 1 thread per core).

Reply:
Anyway, I updated the PR to call the sync method directly. PTAL